File: get_suggestions.py
from google.oauth2 import service_account
from googleapiclient.discovery import build
import os
import json
from datetime import datetime

def handler(request):
    try:
        # Настройки Google Sheets
        SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']
        SERVICE_ACCOUNT_INFO = json.loads(os.environ.get('GOOGLE_SERVICE_ACCOUNT'))
        SPREADSHEET_ID = os.environ.get('SHEET_ID')
        RANGE_NAME = 'Лист1!A:H'
        
        credentials = service_account.Credentials.from_service_account_info(
            SERVICE_ACCOUNT_INFO, scopes=SCOPES)
        service = build('sheets', 'v4', credentials=credentials)
        
        # Получаем данные
        sheet = service.spreadsheets()
        result = sheet.values().get(spreadsheetId=SPREADSHEET_ID,
                                    range=RANGE_NAME).execute()
        values = result.get('values', [])
        
        if not values:
            return {
                'statusCode': 200,
                'headers': {
                    'Content-Type': 'application/json',
                    'Access-Control-Allow-Origin': '*'
                },
                'body': json.dumps([])
            }
        
        # Преобразуем в список словарей
        headers = values[0]
        suggestions = []
        for row in values[1:]:
            if len(row) >= 7:
                suggestion = {
                    'date': row[0] if len(row) > 0 else '',
                    'nick': row[1] if len(row) > 1 else '',
                    'title': row[2] if len(row) > 2 else '',
                    'category': row[3] if len(row) > 3 else '',
                    'link': row[4] if len(row) > 4 else '-',
                    'comment': row[5] if len(row) > 5 else '-',
                    'votes': int(row[6]) if len(row) > 6 and row[6] else 0,
                    'id': row[7] if len(row) > 7 else ''
                }
                suggestions.append(suggestion)
        
        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps(suggestions)
        }
        
    except Exception as e:
        return {
            'statusCode': 500,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({'error': str(e)})
        }

File: api/get-suggestions.js
import { google } from 'googleapis';

export default async function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');
  
  try {
    const auth = new google.auth.GoogleAuth({
      credentials: JSON.parse(process.env.GOOGLE_SERVICE_ACCOUNT),
      scopes: ['https://www.googleapis.com/auth/spreadsheets.readonly'],
    });

    const sheets = google.sheets({ version: 'v4', auth });
    const response = await sheets.spreadsheets.values.get({
      spreadsheetId: process.env.SHEET_ID,
      range: 'Лист1!A:I',
    });

    const rows = response.data.values || [];
    const suggestions = rows.slice(1).map(row => ({
      date: row[0] || '',
      nick: row[1] || '',
      title: row[2] || '',
      category: row[3] || '',
      link: row[4] || '-',
      comment: row[5] || '-',
      votes: parseInt(row[6]) || 0,
      id: row[7] || '',
      avatar: row[8] || ''
    }));

    res.status(200).json(suggestions);
  } catch (error) {
    console.error('Get suggestions error:', error);
    res.status(500).json({ error: error.message });
  }
}
